R/utils.R

```diff
@@ -170,19 +170,28 @@ validate_data_has_expected_cols <- function(data,
 #' @param data A tibble.
 #' @param composite_unique_cols A vector of names of columns that shall be
 #'   unique in their combination.
+#' @param throw_error Boolean, if TRUE error is thrown on failures, otherwise a
+#'   warning.
 #'
 #' @return NULL
 #' @export
-report_all_duplicate_kinds <- function(data, composite_unique_cols) {
-
+report_all_duplicate_kinds <- function(data, composite_unique_cols, throw_error = FALSE) {
   validate_data_has_expected_cols(
     data = data,
     expected_columns = composite_unique_cols
   )
 
-  report_duplicates(data = data, cols = names(data))
+  report_duplicates(
+    data = data,
+    cols = names(data),
+    throw_error = throw_error
+  )
 
-  report_duplicates(data = dplyr::distinct(data), cols = composite_unique_cols) # removed duplicates on all columns
+  report_duplicates(
+    data = dplyr::distinct(data),
```

Comment:
just out of curiosity, if applying `distinct()` to an entire dataset, wouldn't `distinct_all()` be the more natural option?

Reply:
it would, but that would mean I would have to make function report_duplicates more complicated or duplicate it. This way I achieve the same with using just 1 function for checking and reporting